libraries/AP_HAL_ChibiOS/hwdef/TBS-L431-BattMon/hwdef.dat

```diff
@@ -0,0 +1,62 @@
+include ../TBS-L431/hwdef.inc
+
+USE_BOOTLOADER_FROM_BOARD TBS-L431-Periph
+
+define AP_PERIPH_BATTERY_ENABLED 1
+define AP_PERIPH_RELAY_ENABLED 1
+define AP_RELAY_ENABLED 1
+define AP_RELAY_NUM_RELAYS 3
+define AP_DAC_ENABLED 1
+
+define HAL_USE_ADC TRUE
+define STM32_ADC_USE_ADC1 TRUE
+
+PA4 BATT_VOLTAGE_SENS  ADC1 SCALE(1) # VBAT
+PA3 BATT_CURRENT_SENS  ADC1 SCALE(1)
+PA0 BATT2_VOLTAGE_SENS ADC1 SCALE(1) # 5V
+PA1 BATT3_VOLTAGE_SENS ADC1 SCALE(1) # 12V
+PA2 BATT4_VOLTAGE_SENS ADC1 SCALE(1) # ADJ
+
+PA6 BATT2_EN OUTPUT GPIO(50) # 5V
+PA7 BATT3_EN OUTPUT GPIO(51) # 12V
+PA8 BATT4_EN OUTPUT GPIO(52) # ADJ
+
+define RELAY1_PIN_DEFAULT 50
+define RELAY2_PIN_DEFAULT 51
+define RELAY3_PIN_DEFAULT 52
+
+define AP_BATT_MONITOR_MAX_INSTANCES 4
+
+# VBAT
+define HAL_BATT_MONITOR_DEFAULT  4
+define HAL_BATT_VOLT_PIN 9
+define HAL_BATT_VOLT_SCALE 21
+define HAL_BATT_CURR_PIN 8
+define HAL_BATT_CURR_SCALE 37.5
+
+# 5V
+define HAL_BATT2_MONITOR_DEFAULT  3
+define HAL_BATT2_VOLT_PIN 5
+define HAL_BATT2_VOLT_SCALE 2
+
+# disable unnecessary threads
+define HAL_NO_RCOUT_THREAD
+undef HAL_RCIN_THREAD_ENABLED
+define HAL_RCIN_THREAD_ENABLED 0
+
+define AP_DAC_MCP40D1X_ENABLED 1
+define AP_DAC_DEFAULT_ADDR 0x2E
+define AP_DAC_DEFAULT_VREF 20
+define AP_DAC_DEFAULT_BUS  0
+define AP_DAC_DEFAULT_VOLT 10
+define AP_DAC_DEFAULT_TYPE AP_DAC_TYPE_MCP40D1x
+define AP_DAC_CONVERSION_EQ(vo,vr) ((127.0/10000.0) * (0.8 * 21000 * 866 + 0.8 * 21000 * 169000 - (vo - 1.0) * 866 * 169000)/((vo - 1.0) * 169000 - 0.8 * 21000))
```

Comment:
should use () around variables in right size of define

Reply:
Fixed, thanks